clients/src/main/java/org/apache/kafka/common/protocol/MessageUtil.java

```diff
@@ -37,6 +38,17 @@ public static short serializedUtf8Length(CharSequence input) {
         return (short) count;
     }
 
+    public static byte[] byteBufferToArray(ByteBuffer buf) {
```

Comment:
Maybe include a short javadoc here? Might be worth mentioning that won't affect a `mark` on the given buffer